src/Modules/FreeModules-graded.jl

```diff
@@ -1214,25 +1214,32 @@ function direct_product(G::ModuleFP_dec...; task::Symbol = :none)
 end
 ⊕(M::ModuleFP_dec...) = direct_product(M..., task = :none)
 
+function canonical_injections(G::ModuleFP_dec)
+  H = get_attribute(G, :direct_product)
+  @req H !== nothing "module not a direct product"
+  return [canonical_injection(G, i) for i in 1:length(H)]
+end
 
-function Hecke.canonical_injection(G::ModuleFP_dec, i::Int)
+function canonical_injection(G::ModuleFP_dec, i::Int)
   H = get_attribute(G, :direct_product)
-  if H === nothing
-    error("module not a direct product")
-  end
-  0<i<= length(H) || error("index out of bound")
-  j = i == 1 ? 0 : sum(ngens(H[l]) for l=1:i-1) -1
```

Comment:
The `-1` here at the end of this line seems wrong to me, so I removed it. It has been introduced by @fieker in https://github.com/oscar-system/Oscar.jl/commit/ecfbd92a771ba141f50732aa633cfbcf7345a8e6 (a long time ago). Could you please have a look if that has been a typo (that will be fixed by this) or if this introduces faulty behavior?

Reply:
@fieker ping